File: 0229-majority-element-ii/0229-majority-element-ii.cpp
class Solution {
public:
    vector<int> majorityElement(vector<int>& nums) {
        int n = nums.size();
        unordered_map<int, int> frequency;
        vector<int> result;

        for (int num : nums) {
            frequency[num]++;
        }

        for (auto& pair : frequency) {
            if (pair.second > n / 3) {
                result.push_back(pair.first);
            }
        }

        return result;

    }
};

File: 0229-majority-element-ii/0229-majority-element-ii.py
class Solution(object):
    def majorityElement(self, nums):
        """
        :type nums: List[int]
        :rtype: List[int]
        """
        n = len(nums)
        result = []
        frequency = Counter(nums)

        for k , v in frequency.items():
            if v > n//3:
                result.append(k)
        
        # result.sort()
        return result